android/modules/media/src/java/ti/modules/titanium/media/TiCameraXActivity.java

```diff
@@ -417,6 +421,30 @@ private void startCamera()
 		int rotation = getWindowManager().getDefaultDisplay().getRotation();
 
 		Activity activity = TiApplication.getAppCurrentActivity();
+
+		orientationEventListener = new OrientationEventListener(activity)
+		{
+			@Override
+			public void onOrientationChanged(int orientation)
+			{
+				if (orientationEventListener == null || orientation == ORIENTATION_UNKNOWN) {
+					return;
+				}
+				int rotation = getWindowManager().getDefaultDisplay().getRotation();
+				if (lastDisplayOrientation != rotation) {
+					imageCapture.setTargetRotation(rotation);
+					lastDisplayOrientation = rotation;
+				}
+
+			}
+		};
+		if (orientationEventListener.canDetectOrientation()) {
```

Comment:
Wouldn't it make more sense to do the whole constructore only if orientation is detected? That way, you don't need to `null` it again

Reply:
Ah, of course!